### PR TITLE
RGW/Keystone integration improvements

### PR DESCRIPTION
**Pull request by Logan2211**
* Add support for deploying RGW with Keystone user token authentication instead of requiring the deprecated Keystone admin token be enabled.

* Allow disabling the nss PKI database which is not applicable when PKI auth is not used in Keystone.

* Add the ability to configure RGW using the Keystone v3 API.

### CODE REVIEW

File: roles/ceph-common/defaults/main.yml [hunk 1]
```diff
@@ -277,7 +277,14 @@ radosgw_civetweb_bind_ip: "{{ ansible_default_ipv4.address }}"
 radosgw_civetweb_num_threads: 50
 radosgw_keystone: false # activate OpenStack Keystone options full detail here: http://ceph.com/docs/master/radosgw/keystone/
 #radosgw_keystone_url: # url:admin_port ie: http://192.168.0.1:35357
-radosgw_keystone_admin_token: password
+#radosgw_keystone_api_version: 3 #default is 2
```
Comment (leseb) [inline roles/ceph-common/defaults/main.yml]: why are all the variables commented?

**Comment by Logan2211** [comment]
@leseb: Fixed it up. Thanks

**Comment by leseb** [comment]
@Logan2211 I have the impression that ` roles/ceph-common/defaults/main.yml` and group_vars/all.yml.sample` are different, did you re-run `generate_group_vars_sample.sh`?

**Comment by Logan2211** [comment]
Indeed there was a whitespace difference in the group_vars sample. I was just syncing the vars by hand, didn't realize there was a script, and I guess my editor stripped the extra whitespace from the end. Should be gtg now. :)

**Comment by leseb** [comment]
jenkins test ansible2.2-centos7_cluster

**Comment by leseb** [comment]
jenkins test ansible2.2-docker_cluster

**Comment by leseb** [comment]
jenkins test ansible2.2-xenial_cluster

**Comment by Logan2211** [comment]
Was just looking -- this is actually broken now that the commenting is removed from the ceph-common vars. It is supposed to be "either radosgw_keystone_admin_token is defined, or else all of the radosgw_keystone_admin_user/radosgw_keystone_admin_password/radosgw_keystone_admin_tenant vars are defined".. but having them all defaulted will just bypass the user auth stuff all the time.

**Comment by Logan2211** [comment]
Fixed that mess up. Now there is a switch called `radosgw_keystone_auth_method` to switch between admin_token and auth_token.

**Pull request merged**